frontend/components/UpgradeURD.tsx

```diff
@@ -0,0 +1,319 @@
+import React, { useContext, useEffect, useState } from 'react';
+import {
+  Box,
+  Button,
+  Flex,
+  Step,
+  StepDescription,
+  StepIndicator,
+  StepNumber,
+  Stepper,
+  StepSeparator,
+  StepStatus,
+  StepTitle,
+  Text,
+  useColorModeValue,
+  useSteps,
+  useToast,
+  VStack,
+} from '@chakra-ui/react';
+import { WalletContext } from '@/components/wallet/WalletContext';
+import { ethers } from 'ethers';
+import LSP1GraveForwarderAbi from '@/abis/LSP1GraveForwarder.json';
+import { LSP1GraveForwarder } from '@/contracts';
+import { formatAddress } from '@/utils/tokenUtils';
+import { FaCheckCircle } from 'react-icons/fa';
+import {
+  resetLSPDelegates,
+  setForwarderAsLSPDelegate,
+  updateBECPermissions,
+} from '@/utils/urdUtils';
+
+const initialLeavingSteps = [
+  {
+    title: 'Give your 🆙 necessary permissions',
+    instructions: 'Give permissions to your Browser Extension Controller.',
+    instructions2: { text: 'This can also be done manually.', address: null },
+    completeText: { text: 'PERMISSION SET', address: null },
+    complete: false,
+  },
+  {
+    title: 'Reset previous LSP 7 and LSP8 delegate permissions',
+    completeText: { text: '', address: null },
+    complete: false,
+  },
+  {
+    title: 'Migrate your GRAVE to the new forwarder',
+    completeText: { text: '', address: null },
+    complete: false,
+  },
+  {
+    title: 'Direct all 🆙 spam to the GRAVE',
+    completeText: { text: 'SPAM IS DEAD', address: null },
+    complete: false,
+  },
+];
+
+export const UpgradeURD = ({
+  oldForwarderAddress,
+}: {
+  oldForwarderAddress: string;
+}) => {
+  const walletContext = useContext(WalletContext);
+  const bgColor = useColorModeValue('light.green.brand', 'dark.purple.200');
```

Comment:
do you need useColorModeValue if we 'technically' use always the same theme?

Reply:
As discussed we'll keep this and remove it later in one go